### PR TITLE
bugfix: don't pass cell children to useQuery

### PR DESCRIPTION
**Pull request by wminshew**
Hi - first PR here, related to issue https://github.com/redwoodjs/redwood/issues/1583

Very open to feedback :) JS changes feel straightforward, though wasn't sure if more could/should be done to verify that the remaining `variables` are appropriate for `useQuery`. Am a novice when it comes to typescript, so did my best but imagine a cleaner solution exists. Opening this to start the conversation!

one quick question I had -- I used `eslint-disable-line` but it wasn't clear to me why `{ children: _children }` was still triggering an error based on the `varsIgnorePattern` from the `eslint-config` pasted below

```
'no-unused-vars': [
      'error',
      { varsIgnorePattern: '^_', argsIgnorePattern: '^_' },
    ],
```

Last -- I was able to successfully run `cypress` tests locally, but couldn't seem to get a redwood repo running with my local version of redwood (macos 10.15.5); (I suspect, based on the below error, it is something very silly I'm forgetting to do..)

image

### CODE REVIEW

**Comment by thedavidprice** [comment]
Thanks for this @wminshew! I've looped in @peterp for review.

@dac09 could you also make a recommendation about which release this PR should target?

**Comment by Tobbe** [comment]
Great first PR Will! Thank you!

In the screenshot you posted I see you're still using `<RedwoodProvider>`. If you're working on a somewhat fresh `main` branch you need to switch to `<RedwoodApolloProvider>` instead. See the release notes for the latest RW release for details

**Comment by wminshew** [comment]
> In the screenshot you posted I see you're still using `<RedwoodProvider>`. If you're working on a somewhat fresh `main` branch you need to switch to `<RedwoodApolloProvider>` instead. See the release notes for the latest RW release for details

ah thank you that makes sense. I tried 1/ the [example-invoice](https://github.com/redwoodjs/example-invoice) repo & also my own older (by a few weeks?) repo from the tutorial. I will log the issue in the `example-invoice` repo for now

**Comment by Tobbe** [comment]
I know @peterp said to do this

```js
export const beforeQuery = ({ children, ...variables }) => {
  return { variables }
}
```

But looking at it I think it would make more sense to just never send child nodes to `beforeQuery` in the first place. So I'd suggest this instead, and not change the signature of `beforeQuery`:

```jsx

  return (props: Record<string, unknown>) => {
    const { children, ...variables } = props

    return (
      <Query
        query={
          typeof QUERY === 'function' ? QUERY(beforeQuery(variables)) : QUERY
        }
        {...beforeQuery(variables)}
      >
```

But I'd wait for Peter's input before making any changes

File: packages/web/src/components/withCellHOC.tsx [hunk 1]
```diff
@@ -26,7 +26,6 @@ export type CellSuccessStateComponent =
 
 export interface WithCellProps {
   beforeQuery?: <TProps>(props: TProps) => { variables: TProps }
-  // @ts-expect-error We do not know, and even really care, what they are here.
```
Comment (wminshew) [inline packages/web/src/components/withCellHOC.tsx:29]: this was throwing an "unused directive" error for me so I removed it -- please let me know if you want it back in

**Comment by wminshew** [comment]
> But I'd wait for Peter's input before making any changes

ah.. sorry, I thought what you said made sense and I actually missed this last line & went ahead with the changes. Not a problem for me to revert pending Peter's feedback!

**Comment by Tobbe** [comment]
> ah.. sorry, I thought what you said made sense

Happy to hear it wasn't just me it made sense to 😁

File: packages/web/src/components/withCellHOC.tsx (hunk 1 above)
Comment (wminshew) [reply packages/web/src/components/withCellHOC.tsx]: looks like this caused some issues with the automated lintings/builds; reverted

Comment (Tobbe) [reply packages/web/src/components/withCellHOC.tsx]: You can tighten up the typing, that way you can remove the comment
```
  QUERY: DocumentNode | ((variables: Record<string, unknown>) => DocumentNode)
```

Comment (wminshew) [reply packages/web/src/components/withCellHOC.tsx:29]: good call -- updated

**Review by peterp — APPROVED** [review]

**Comment by peterp** [comment]
LGTM thanks @wminshew and @Tobbe

**Comment by Tobbe** [comment]
@dac09 Are we OK to merge this? Or does it interfere with your prerender stuff?

**Comment by dac09** [comment]
Looks fine to me, I can handle the merge conflicts :)

**Pull request merged**